Common++/header/IpUtils.h

```diff
@@ -14,6 +14,11 @@
 #include <winsock2.h>
 #include <ws2tcpip.h>
 #endif
+#ifdef FREEBSD
+#include <sys/socket.h>

```

Comment:
do we need `sys/socket.h` in the header file or can we move it to `IpUtils.cpp` file?

Reply:
In FreeBSD, struct sockaddr is defined in sys/socket.h and forward declared in netinet/in.h
Also, AF_* macro is defined in sys/socket.h.

If I follow your suggestion, it breaks the code below which include `IpUtils.h`

```
./Common++/src/IpAddress.cpp:42:	if (inet_pton(AF_INET, addressAsString, &ip4Addr) != 0)
./Common++/src/IpAddress.cpp:46:	else if (inet_pton(AF_INET6, addressAsString, &ip6Addr) != 0)
./Common++/src/IpAddress.cpp:75:	if (inet_ntop(AF_INET, m_pInAddr, m_AddressAsString, MAX_ADDR_STRING_LEN) == 0)
./Common++/src/IpAddress.cpp:85:	if (inet_ntop(AF_INET, m_pInAddr, m_AddressAsString, MAX_ADDR_STRING_LEN) == 0)
./Common++/src/IpAddress.cpp:99:	if (inet_pton(AF_INET, addressAsString , m_pInAddr) == 0)
./Common++/src/IpAddress.cpp:192:	if (inet_pton(AF_INET6, addressAsString , m_pInAddr) == 0)
./Common++/src/IpAddress.cpp:207:	if (inet_ntop(AF_INET6, m_pInAddr, m_AddressAsString, MAX_ADDR_STRING_LEN) == 0)
./Examples/KniPong/main.cpp:270:	if ((sock.m_Socket = socket(AF_INET, SOCK_DGRAM, 0)) == INVALID_FD)
./Examples/KniPong/main.cpp:278:	egress.sin_family = AF_INET;
./Examples/KniPong/main.cpp:415:	ingress.sin_family = AF_INET;
./Pcap++/src/PcapLiveDevice.cpp:676:	int socketfd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
./Pcap++/src/PcapLiveDevice.cpp:732:	int socketfd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
./Pcap++/src/RawSocketDevice.cpp:380:	int family = (m_SockFamily == IPv4 ? AF_INET : AF_INET6);
./Pcap++/src/RawSocketDevice.cpp:414:		int res = inet_pton(AF_INET6, m_InterfaceIP->toString().c_str(), &localAddrIPv6.sin6_addr.s6_addr);
./Pcap++/src/RawSocketDevice.cpp:474:			if  (curAddr->ifa_addr->sa_family == AF_INET)
./Pcap++/src/RawSocketDevice.cpp:485:			else if (curAddr->ifa_addr->sa_family == AF_INET6)
```

See [FreeBSD doc](https://www.freebsd.org/doc/en_US.ISO8859-1/books/developers-handbook/sockets-essential-functions.html)

Reply:
Got it, thanks!